tests/unit/anchore_engine/test_utils.py

```diff
@@ -92,6 +99,58 @@ def test_parse_dockerimage_string(image, expected):
     assert result == expected
 
 
+def test_run_sanitize_good_input():
+    # Setup input
+    input_cmd_list = ["wc", "-l"]
+
+    # Function under test
+    output_cmd_list = run_sanitize(input_cmd_list)
+
+    # Validate input
+    output_cmd_list == input_cmd_list
+
+
+@pytest.mark.parametrize(
+    "input",
+    [";&<>", ";", "&", "<", ">"],
+)
+def test_run_sanitize_bad_input(input):
+    with pytest.raises(Exception):
+        run_sanitize(input)
+
+
+@pytest.mark.parametrize(
+    "cmds_list, expected_return_code, expected_stdout, expected_stderr",
+    [([], None, None, None), ([["pwd"], ["wc", "-l"]], 0, "1", b"")],
+)
+def test_run_piped_command(
+    cmds_list, expected_return_code, expected_stdout, expected_stderr
+):
+    # Function under test
+    return_code, stdout, stderr = run_piped_command_list(cmds_list)
+
+    # Binary string returned in different environments can be padded with different amounts of whitespace
+    # So where expected_output != None, convert it to utf-8 and trim it so we get a clean, reliable comparison
+    if stdout is not None:
+        stdout = stdout.decode("utf-8").strip()
+
+    # Validate input
+    assert return_code == expected_return_code
+    assert stdout == expected_stdout
+    assert stderr == expected_stderr
+
+
+def test_run_invalid_piped_command():
+    # Setup input var
+    first_command = shlex.split("pwd")
+    bad_command = shlex.split("<")
+    cmds_list = [first_command, bad_command]
+
+    with pytest.raises(Exception):
```

Comment:
if this is catching any exception then at least the message should be verified, otherwise it might be too broad - making this test not that useful (false positive)

Reply:
I will verify the message, but `Exception` is what the function under test raises, and it is used in enough places that I don't want to pull the testing involved on changing that into this change.

Reply:
Yeah, if the underlying code raises `Exception` this is going to be more involved. Bummer :(